arangod/IResearch/IResearchLink.cpp

```diff
@@ -248,16 +248,58 @@ class IResearchFlushSubscription final : public FlushSubscription {
   std::atomic<TRI_voc_tick_t> _tick;
 };
 
-bool readTick(irs::bytes_ref const& payload, TRI_voc_tick_t& tick) noexcept {
+enum class SegmentPayloadVersion : irs::byte_type {
+  // Only max tick stored. Possibly has uncommitted WAL ticks
+  SingleTick = 0,
+  // Two stage commit ticks are stored.
+  // Min tick is fully committed but between first and second tick
+  // uncommitted WAL records may be present. After second tick nothing is
+  // committed.
+  TwoStageTick = 1
+};
+
+bool readTick(irs::bytes_ref const& payload, TRI_voc_tick_t& tickLow,
+              TRI_voc_tick_t& tickHigh) noexcept {
   static_assert(sizeof(uint64_t) == sizeof(TRI_voc_tick_t));
 
-  if (payload.size() != sizeof(uint64_t)) {
+  if (payload.size() < sizeof(uint64_t)) {
+    LOG_TOPIC("41474", ERR, TOPIC) << "Unexpected segment payload size "
+                                   << payload.size() << " for initial check";
     return false;
   }
-
-  std::memcpy(&tick, payload.c_str(), sizeof(uint64_t));
-  tick = TRI_voc_tick_t(irs::numeric_utils::ntoh64(tick));
-
+  std::memcpy(&tickLow, payload.c_str(), sizeof(uint64_t));
+  tickLow = TRI_voc_tick_t(irs::numeric_utils::ntoh64(tickLow));
+  tickHigh = std::numeric_limits<TRI_voc_tick_t>::max();
+  SegmentPayloadVersion version{SegmentPayloadVersion::SingleTick};
+  if (payload.size() > sizeof(uint64_t) + sizeof(version)) {
+    std::memcpy(&version, payload.c_str() + sizeof(uint64_t), sizeof(version));
```

Comment:
Do we need to `ntoh` the version here to make it portable? Or is the version supposed to be a single byte only? If so, please leave a code comment here that explains that.

Reply:
Version is 1 byte, Will leave a comment

Reply:
But no. Other versions are uint32_t so let it be so. And +ntoh :)